src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -37,3 +47,19 @@ def validate_reservations_summaries(namespace):
         raise CLIError("usage error: --grain  can be either daily or monthly.")
     if data_grain == 'daily' and (not namespace.start_date or not namespace.end_date):
         raise CLIError("usage error: Both --start-date and --end-date need to be supplied for daily grain.")
+
+
+def validate_budget_parameters(namespace):
+    """lowercase the time grain for comparison"""
+
+    budgetcategory = namespace.category.lower()
+    if budgetcategory != 'cost' and budgetcategory != 'usage':
+        raise CLIError("usage error: --parameters.category must be set to either cost or usage")
+
+    time_grain = namespace.time_grain.lower().strip()
+
+    if time_grain != 'annually' and time_grain != 'quarterly' and time_grain != 'monthly':
+        raise CLIError("usage error: --parameters.time_grain can be 'Annually', 'Quarterly', or 'Monthly'.")
+
+    if namespace.amount < 0:
+        raise CLIError("usage error: --parameters.time_period.amount must be greater than 0")
```

Comment:
There is no such option `--parameter.time_period.amount`.

Reply:
Fixed typo by removing time_period